### PR TITLE
Refactor Flavor data model

### LINKED ISSUE #951: Refactor Flavor data model
The [Flavor](https://github.com/eclipse-xpanse/xpanse/blob/145070f719637cba655176f1f0fc701b28ae1c73/modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/Flavor.java#L19) type is currently shared between service template and orderable services. This must be changed. 

- [ ] Remove `properties` field from Flavor type and use it in the `UserOrderableServiceVo` type. 
- [ ] Create a new type `FlavorWithProperties` - which extends the Flavor type and additionally has `properties` field. Use it in the `ServiceTemplateDetailVo` type. 
- [ ] Check and update UI.

### PR DESCRIPTION
**Pull request by baixinsui**
fixed #951 
related https://github.com/eclipse-xpanse/xpanse-ui/pull/230

### CODE REVIEW

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/utils/DeployEnvironments.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
 import org.eclipse.xpanse.modules.models.service.common.enums.Csp;
 import org.eclipse.xpanse.modules.models.service.deploy.exceptions.FlavorInvalidException;
 import org.eclipse.xpanse.modules.models.servicetemplate.DeployVariable;
-import org.eclipse.xpanse.modules.models.servicetemplate.Flavor;
+import org.eclipse.xpanse.modules.models.servicetemplate.FlavorWithProperties;
```
Comment (iskey) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/utils/DeployEnvironments.java:20]: I feel like keeping the original `Flavor` is better, with adding a new class named `FlavorWithoutProperties`/`FlavorBasic`.

**Review by iskey — CHANGES_REQUESTED** [review]
comments added

**Review by swaroopar — CHANGES_REQUESTED** [review]
We can go with the suggestion from Yichao.. FlavorBasic is fine for me.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/utils/DeployEnvironments.java (hunk 1 above)
Comment (baixinsui) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/utils/DeployEnvironments.java:20]: Ok. I added  a new class `FlavorBasic` without the field `properties` for user querying.

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceCatalogApi.java [hunk 2]
```diff
@@ -166,7 +167,13 @@ private UserOrderableServiceVo convertToUserOrderableServiceVo(
             userOrderableServiceVo.setDescription(
                     serviceTemplateEntity.getOcl().getDescription());
             userOrderableServiceVo.setBilling(serviceTemplateEntity.getOcl().getBilling());
-            userOrderableServiceVo.setFlavors(serviceTemplateEntity.getOcl().getFlavors());
+            List<FlavorBasic> flavorBasics = serviceTemplateEntity.getOcl().getFlavors()
+                    .stream().map(flavorWithProperties -> {
+                        FlavorBasic flavorBasic = new FlavorBasic();
+                        BeanUtils.copyProperties(flavorWithProperties, flavorBasic);
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceCatalogApi.java:173]: @baixinsui flavorWithProperties must be just flavor now?

Comment (baixinsui) [reply modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceCatalogApi.java]: Updated.

**Review by iskey — APPROVED** [review]
LGTM

**Review by swaroopar — APPROVED** [review]

**Pull request merged**